src/Polysemy/Internal/Combinators.hs

```diff
@@ -381,5 +424,151 @@ transform
 transform f (Sem m) = Sem $ \k -> m $ \u ->
   k $ hoist (transform f) $ case decomp u of
     Left g -> g
-    Right (Weaving e s wv ex ins) ->
-      injWeaving (Weaving (f e) s wv ex ins)
+    Right (Weaving e mkT lwr ex) ->
+      injWeaving (Weaving (f e) mkT lwr ex)
+
+
+-- | An effect for running monadic actions within a higher-order effect
+-- currently being interpreted.
+newtype RunH z (m :: * -> *) a where
+  RunH :: z a -> RunH z m a
+
+-- | Run a monadic action given by a higher-order effect that is currently
+-- being interpreted.
+--
+-- @since TODO
+runH :: Member (RunH z) r => z a -> Sem r a
```

Comment:
Does this magically ensure that each following HO action you run "inherits" the state of the previous one, as opposed to the `*Simple` variants, where you're still threading it manually?

Reply:
Yes.